include/zephyr/drivers/sensor/stmemsc.h

```diff
@@ -0,0 +1,80 @@
+/* ST Microelectronics STMEMS hal i/f
```

Comment:
Did you move the file under drivers/sensor/stmemsc here?
If so, is it exactly the same (seems something is missing)?
Why didn't you remove the original file?
Was it a request during review?

Reply:
Also, sorry if I didn't review it before...

Reply:
Can you also please merge the two commits together?

Reply:
The contents is indeed out-of-date; I had previously copied it (for now) to avoid depending on a relative path in the CMakeLists file. As I recall, nordicjm identified that specific issue with their review but it seems to have since been dismissed. If it is no longer a problem I can put things back and follow the pattern used by the other ST drivers (like below).

If this PR is merged, the CMakeLists for other ST drivers could instead rely on that header being in the include path and remove the relative path (below), though if I move the file right now, the sources in `drivers/sensor/stmemsc` may no longer compile without a corresponding change to to use the search path rather than assuming the header is in the same directory. 

Please let me know which way you prefer and I'll make the necessary change. I'll get the commits combined.

https://github.com/zephyrproject-rtos/zephyr/blob/3b36911461a49226868dab4a63093f651eca4cc5/drivers/sensor/lis2dw12/CMakeLists.txt#L12

Reply:
> The contents is indeed out-of-date; I had previously copied it (for now) to avoid depending on a relative path in the CMakeLists file. As I recall, nordicjm identified that specific issue with their review but it seems to have since been dismissed. If it is no longer a problem I can put things back and follow the pattern used by the other ST drivers (like below).

Yes, I would prefer to go as the other drivers. Maybe we can move this filer later on in another PR taking care of all the drivers

Reply:
Sounds good. I made the change.